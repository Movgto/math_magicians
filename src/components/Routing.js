import {
  Route, Routes, HashRouter, NavLink,
} from 'react-router-dom';
import App from '../App';
import Quotes from './Quotes';
import Home from './Home';

const apiKey = "aSn23Z2Z26ES8eaiczrw2w==SHUDji9fSQZwMG1g";

const Routing = () => (
  <>
    <HashRouter>
      <nav>
        <ul>
          <li><NavLink className="link" to="/">Home</NavLink></li>
          <li><NavLink className="link" to="/calculator">Calculator</NavLink></li>
          <li><NavLink className="link" to="/quote">Quote</NavLink></li>
        </ul>
      </nav>
      <Routes>
        <Route path="/">
          <Route index element={<Home />} />
          <Route path="/calculator" element={<App />} />
          <Route path="/quote" element={<Quotes apiKey={apiKey} data-testid="quotes" />} />
        </Route>
      </Routes>
    </HashRouter>
  </>
);

export default Routing;