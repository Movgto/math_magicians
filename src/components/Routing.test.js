import { render, screen, act, findByTestId } from '@testing-library/react';
import Routing from './Routing';

describe('Navigation: tests proper navigation behavior when clicking links', () => {
  test('Start at home page', () => {
    render(<Routing />);
    const expected = 'Welcome to our page!';
    const welcomeMsg = screen.getByText(expected).textContent;
    expect(welcomeMsg).toBe(expected);
  });

  test('Navigation to Calculator page', () => {
    render(<Routing />);
    const calculatorLink = screen.getByText('Calculator');
    act(() => {
      calculatorLink.click()
    });
    screen.getByTestId('calculator');
  })

  test('Navigation to Quote page', () => {
    render(<Routing />);
    const quoteLink = screen.getByText('Quote');
    act(() => {
      quoteLink.click();
    });
    screen.getByTestId('quotes');
  });
});