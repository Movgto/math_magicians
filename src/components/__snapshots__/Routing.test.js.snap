// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot renders properly 1`] = `
Array [
  <nav>
    <ul>
      <li>
        <a
          aria-current="page"
          className="link active"
          href="#/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="link"
          href="#/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          className="link"
          href="#/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>,
  <div
    id="home"
  >
    <h1>
      Welcome to our page!
    </h1>
    <p>
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Laudantium asperiores cumque dignissimos esse iste suscipit, non voluptatibus, perferendis repellat dolores numquam quasi. Earum nam velit possimus quisquam accusamus mollitia iure!
    </p>
  </div>,
]
`;
